src/libxtp/calculators/kmcmultiple.cc

```diff
@@ -284,6 +286,12 @@ void KMCMultiple::RunVSSM(ctp::Topology *top)
                 avgdiffusiontensor += (_carriers[i]->dr_travelled)|(_carriers[i]->dr_travelled);
             }
         }
+        
+        if(step!=0 && step%(int(1e9))==0){
+            unsigned long diffusionsteps=step/diffusionresolution;
+            tools::matrix result=avgdiffusiontensor/(diffusionsteps*2*simtime*_numberofcharges);
+            cout<<endl<<"Step: "<<step<<" Diffusion tensor averaged over all carriers (nm^2/s):" << endl << result << endl;
```

Comment:
Can we have the same for average velocity, useful in mobility calculation

Reply:
yes, okay